tasks/pkg-debian/install-apm-inject.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Set APM injection languages and versions for Debian
+  set_fact:
+    agent_dd_apm_install_pkgs: "{{ (agent_dd_apm_install_pkgs |
+      default([], true)) + ['datadog-apm-library-' + (item | regex_replace('[:]', '='))] }}"
+  loop: "{{ pre_processed_datadog_apm_libraries }}"
+
+- name: Install APM Inject Library
+  apt:
+    name: 'datadog-apm-inject'
+    state: latest # noqa package-latest
+    update_cache: true
+    cache_valid_time: "{{ datadog_apt_cache_valid_time }}"
+
+- name: Install APM inject libraries
+  apt:
+    name: "{{ (item | default([], true)) }}"
+    state: "{{ 'present' if '=' in item else 'latest' }}" # noqa package-latest
+  loop: "{{ agent_dd_apm_install_pkgs }}"
+  when: not ansible_check_mode
+
+- name: Show installed tracers
+  debug:
+    msg: "Installed tracers: {{ agent_dd_apm_install_pkgs }}"
+
+- name: Prevent libraries from being upgraded
+  dpkg_selections:
+    name: "{{ item.split(':')[0] }}"
+    selection: hold
+  loop: "{{ pre_processed_datadog_apm_libraries }}"
```

Comment:
Let's use `when: not ansible_check_mode` here as well.

Reply:
Did you actually test this? It doesn't work for me, saying:
```
TASK [ansible-datadog : Prevent libraries from being upgraded] ***************************************************
failed: [localhost] (item=python) => {"ansible_loop_var": "item", "changed": false, "item": "python", "msg": "Failed to find package 'python' to perform selection 'hold'."}
```

when I use `datadog_apm_instrumentation_libraries: ["python"]`. Also, I think we only want to use `hold` when a specific version is provided by the user, but not when we're installing latest (in that case I think we want to leave it unpinned, as we do in the install_script).

Reply:
Fixed this. I realize I was using the wrong variable.